frontend/packages/knative-plugin/src/catalog/useKameletsProvider.ts

```diff
@@ -18,17 +18,15 @@ const normalizeKamelets = (
     const {
       kind,
       metadata: { uid, name, creationTimestamp, annotations },
-      spec: {
-        definition: { title, description },
-      },
+      spec,
     } = k;
     const provider = annotations?.[CAMEL_K_PROVIDER_ANNOTATION] || '';
     const iconUrl = getEventSourceIcon(kind, k);
     const href = `/catalog/ns/${namespace}/eventsource?sourceKind=${CamelKameletBindingModel.kind}&name=${name}`;
     return {
       uid,
-      name: title,
-      description,
+      name: spec?.definition?.title || name,
```

Comment:
I see, there isn't any real backend validation happening for spec or other attributes inside spec. So the optional chaining is added here :+1: